File: benchmark/c/cpw/sum_of_all_proper_divisors_of_a_natural_number/sum_of_all_proper_divisors_of_a_natural_number.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int sum_of_all_proper_divisors_of_a_natural_number ( int num ) ;
int sum_of_all_proper_divisors_of_a_natural_number ( int num ) {
  int result = 0;
  for ( int i = 2;
  i <= sqrt ( num );
  i ++ ) {
    if ( num % i == 0 ) {
      if ( i == ( num / i ) ) result += i;
      else result += ( i + num / i );
    }
  }
  return ( result + 1 );
}

File: translation_result/else_bms/fail/c/cpw/sum_of_all_proper_divisors_of_a_natural_number/gemini/Hinted/sum_of_all_proper_divisors_of_a_natural_number.rs

fn sum_of_all_proper_divisors_of_a_natural_number(num: i32) -> i32 {
    let mut result = 0;
    for i in 2..=(num as f64).sqrt() as i32 {
        if num % i == 0 {
            if i == num / i {
                result += i;
            } else {
                result += i + num / i;
            }
        }
    }
    result + 1
}
